plugins/modules/zos_backup_restore.py

```diff
@@ -0,0 +1,841 @@
+        also be used for temporary data sets created during restore process.
+      - When I(operation=backup), specifies the storage class to use for temporary data sets
+        created during backup process.
+      - If neither of I(sms_storage_class) or I(sms_management_class) are specified, the z/OS
+        system's Automatic Class Selection (ACS) routines will be used.
+    type: str
+    required: False
+  sms_management_class:
+    description:
+      - When I(operation=restore), specifies the management class to use. The management class
+        will also be used for temporary data sets created during restore process.
+      - When I(operation=backup), specifies the management class to use for temporary data sets
+        created during backup process.
+      - If neither of I(sms_storage_class) or I(sms_management_class) are specified, the z/OS
+        system's Automatic Class Selection (ACS) routines will be used.
+    type: str
+    required: False
+  space:
+    description:
+      - If I(operation=backup), specifies the amount of space to allocate for the backup.
+        Please note that even when backing up to a UNIX file, backup contents will be temporarily
+        held in a data set.
+      - If I(operation=backup), specifies the amount of space to allocate for data sets temporarily
+        created during the restore process.
+      - The unit of space used is set using I(space_type).
+      - When I(full_volume=True), I(space) defaults to C(1), otherwise default is C(25)
+    type: int
+    required: false
+    aliases:
+      - size
+  space_type:
+    description:
+      - The unit of measurement to use when defining data set space.
+      - Valid units of size are C(K), C(M), C(G), C(CYL), and C(TRK).
+      - When I(full_volume=True), I(space_type) defaults to C(G), otherwise default is C(M)
+    type: str
+    choices:
+      - K
+      - M
+      - G
+      - CYL
+      - TRK
+    required: false
+    aliases:
+      - unit
+  hlq:
+    description:
+      - Specifies the new HLQ to use for the data sets being restored.
+      - Defaults to running user's username.
+    type: str
+    required: false
+"""
+
+RETURN = r""""""
+
+EXAMPLES = r"""
+- name: Backup all data sets matching the pattern USER.** to data set MY.BACKUP
+  zos_backup_restore:
+    operation: backup
+    data_sets:
+      include: user.**
+    backup_name: MY.BACKUP
+
+- name: Backup all data sets matching the patterns USER.** or PRIVATE.TEST.*
+    excluding data sets matching the pattern USER.PRIVATE.* to data set MY.BACKUP
+  zos_backup_restore:
+    operation: backup
+    data_sets:
+      include:
+        - user.**
+        - private.test.*
+      exclude: user.private.*
+    backup_name: MY.BACKUP
+
+- name: Backup all datasets matching the pattern USER.** to UNIX file /tmp/temp_backup.dzp, ignore recoverable errors.
+  zos_backup_restore:
+    operation: backup
+    data_sets:
+      include: user.**
+    backup_name: /tmp/temp_backup.dzp
```

Comment:
Great questions:
1. dzip extension is completely optional, no extension is enforced. This behavior can be changed if we want. Compression is a simple terse
2. I think it would be good to add to the description that the created “zips” can be used on another system with simple terse and restore

Reply:
Documentation has been added to explicitly mention the lack of naming constraints.